### PR TITLE
feat: add C implementation for `stats/base/dists/arcsine/entropy`

### LINKED ISSUE #3378: [RFC]: Add C implementation for `@stdlib/stats/base/dists/arcsine/entropy`
### Description

This RFC proposes adding a C implementation, including native Node.js addon, C benchmarks and C examples, for the `@stdlib/stats/base/dists/arcsine/entropy` package.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/arcsine/entropy" make install-node-addons
```

#### Run C Examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/arcsine/entropy/examples/c/example.c"
```

#### Run C benchmarks 

````bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/arcsine/entropy/benchmark/c/benchmark.c"
````

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by aayush0325**
Resolves #3378 

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for `@stdlib/stats/base/dists/arcsine/entropy`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3378 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by aayush0325** [comment]
good evening, please give this a review @Planeshifter!

File: lib/node_modules/@stdlib/stats/base/dists/arcsine/entropy/src/main.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+* double y = stdlib_base_arcsine_entropy( 4.0, 12.0 );
+* // returns ~1.838
+*/
+double stdlib_base_dists_arcsine_entropy( const double a, const double b ) {
+	if (
+		stdlib_base_is_nan( a ) ||
+		stdlib_base_is_nan( b ) ||
+		a >= b
+	) {
+		return 0.0 / 0.0; //NaN
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/arcsine/entropy/src/main.c]:
```suggestion
		return 0.0 / 0.0; // NaN
```

File: lib/node_modules/@stdlib/stats/base/dists/arcsine/entropy/README.md [hunk 2]
```diff
@@ -132,6 +132,101 @@ for ( i = 0; i < 10; i++ ) {
...
+        return min + ( v * ( max - min ) );
+    }
+    
+    for ( i = 0; i < 25; i++ ) {
+        a = random_uniform( 0, 20 );
+        b = random_uniform( 0, 20 ) + a;
+        y = stdlib_base_dists_arcsine_entropy( a, b );
+        printf( "x: %lf, a: %lf, b: %lf, h(X;a,b): %lf\n", a, b, y );
+    }
+}
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/arcsine/entropy/README.md]:
```suggestion

static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}

int main( void ) {
    double a;
    double b;
    double y;
    int i;

    for ( i = 0; i < 25; i++ ) {
        a = random_uniform( 0, 20 );
        b = random_uniform( 0, 20 ) + a;
        y = stdlib_base_dists_arcsine_entropy( a, b );
        printf( "x: %lf, a: %lf, b: %lf, h(X;a,b): %lf\n", a, b, y );
    }
}
```

File: lib/node_modules/@stdlib/stats/base/dists/arcsine/entropy/benchmark/benchmark.native.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+var entropy = tryRequire( resolve( __dirname, './../lib/native.js' ) );
+var opts = {
+	'skip': ( entropy instanceof Error )
+};
+
+
+// MAIN //
+
+bench( pkg, opts, function benchmark( b ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/arcsine/entropy/benchmark/benchmark.native.js]:
```suggestion
bench( pkg+'::native', opts, function benchmark( b ) {
```

**Review by Planeshifter — APPROVED** [review]
Thanks! Let's get this in shortly.

**Pull request merged**